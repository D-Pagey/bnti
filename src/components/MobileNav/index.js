import React, { useState } from 'react';
import ClosedNav from './components/ClosedNav';
import OpenNav from './components/OpenNav';

const MobileNav = () => {
    const [isOpen, setIsOpen] = useState(false);

    const toggleOpen = () => {
        setIsOpen(!isOpen);
    };

    return isOpen ? <OpenNav toggleOpen={toggleOpen} /> : <ClosedNav toggleOpen={toggleOpen} />;
};

export default MobileNav;
