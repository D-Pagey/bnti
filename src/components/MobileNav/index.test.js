import React from 'react';
import userEvent from 'user-event';
import MobileNav from '.';

describe('MobileNav component', () => {
    it('should render', () => {
        const { container } = renderWithRouter(<MobileNav />);
        expect(container.firstChild).toMatchSnapshot();
    });

    it('should render OpenNav on Click', () => {
        const { getByTestId } = renderWithRouter(<MobileNav />);
        const menuButton = getByTestId('menuButton');

        userEvent.click(menuButton);
        getByTestId('openNav');
    });
});

/* Qs for Nav
 * expect Closed Nav by default?
 * simulate click, expect openNav but check that closedNav not rendering
 * why user-event over FireEvent?
 */
