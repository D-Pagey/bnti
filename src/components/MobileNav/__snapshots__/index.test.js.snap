// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MobileNav component should render 1`] = `
.c0 {
  background-color: #00A7CB;
  border-top: 2px solid #353535;
  bottom: 0;
  height: 75px;
  left: 0;
  position: fixed;
  right: 0;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 1.25rem;
  height: 100%;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  list-style: none;
  margin-top: 0;
  margin-right: 0;
  margin-bottom: 0;
  margin-left: 0;
  padding-top: 0;
  padding-right: 0;
  padding-bottom: 0;
  padding-left: 0;
  color: black;
}

<div
  class="c0"
  data-testid="closedNav"
>
  <ul
    class="c1"
  >
    <li>
      Home
    </li>
    <li>
      <button
        data-testid="menuButton"
        type="button"
      >
        Menu
      </button>
    </li>
    <li>
      Account
    </li>
  </ul>
</div>
`;
